src/compiler/performance.ts

```diff
@@ -180,13 +180,13 @@ export function enable(system: System = sys) {
     if (!enabled) {
         enabled = true;
         perfHooks ||= tryGetNativePerformanceHooks();
-        if (perfHooks) {
+        if (perfHooks?.performance) {
             timeorigin = perfHooks.performance.timeOrigin;
             // NodeJS's Web Performance API is currently slower than expected, but we'd still like
             // to be able to leverage native trace events when node is run with either `--cpu-prof`
             // or `--prof`, if we're running with our own `--generateCpuProfile` flag, or when
             // running in debug mode (since its possible to generate a cpu profile while debugging).

```

Comment:
Node has a performance API, and its performance API is the same as the web API and is also exposed globally. I'm not sure that checking for the globals actually makes much sense.

![image](https://github.com/microsoft/TypeScript/assets/5341706/e175f253-0467-4248-a40a-20aafc3a4493)

But, I've kept in this PR the behavior of still enabling native perf when the global is present.

Reply:
It turns out that having native perf enabled _is_ very slow. I've made the change to stop doing this on node again (it was probably an oversight) and that caused a very significant perf boost in Node 20.